### ISSUE TITLE
error raised in core.py 

### ISSUE THREAD

**Issue by GaelC92** [MatthiasValvekens/pyHanko#240]
**Describe the bug**
timestamp validation with freetsa, in the context of a visible signature based on a PKCS11 token raises an exception @ line 1721 of /lib/python3.11/site-packages/asn1crypto/core.py
 ValueError: invalid literal for int() with base 10: ''

I catched the exception, as below
        if force:
            native = self.native
            self.contents = None
            try:
                self.set(native)
            except:
                print(self)

It just printed empty lines. 

This did not happen before I attempted to add visible signature. 

How to reproduce : 
it will may not be the easiest to reproduce : 
- using mac os X
- setting arch to i386_x64
- using a PKCS11 eid (AC is Certeurope http://www.certeurope.fr/reference/eid_root.crt / http://www.certeurope.fr/reference/eid_user.crt)
- signature LTV based on doc example, using freetsa root AC for validation.
- adapted to use pdf appearance 
everything works fine, except there seem to be a faulty input somewhere that is not handled in this specific case 

**Expected behavior**
uncertain. 
It may be the expected behavior, in which case some useful error message would be welcome.

**Screenshots**
here is a stack dump 
Traceback (most recent call last):
  File "/Users/jdoe/project-x86/src/signature.py", line 128, in <module>
    pdf_signer.sign_pdf(
  File "/Users/jdoe/project-x86/lib/python3.11/site-packages/pyhanko/sign/signers/pdf_signer.py", line 1408, in sign_pdf
    result = asyncio.run(
             ^^^^^^^^^^^^
  File "/usr/local/Homebrew/Cellar/python@3.11/3.11.2_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/runners.py", line 190, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/usr/local/Homebrew/Cellar/python@3.11/3.11.2_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/Homebrew/Cellar/python@3.11/3.11.2_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Users/jdoe/project-x86/lib/python3.11/site-packages/pyhanko/sign/signers/pdf_signer.py", line 1500, in async_sign_pdf
    await post_signing_doc.post_signature_processing(
  File "/Users/jdoe/project-x86/lib/python3.11/site-packages/pyhanko/sign/signers/pdf_signer.py", line 2601, in post_signature_processing
    validation.DocumentSecurityStore.supply_dss_in_writer(
  File "/Users/jdoe/project-x86/lib/python3.11/site-packages/pyhanko/sign/validation/dss.py", line 464, in supply_dss_in_writer
    dss.register_vri(
  File "/Users/jdoe/project-x86/lib/python3.11/site-packages/pyhanko/sign/validation/dss.py", line 231, in register_vri
    cert_refs.update(set(self._embed_certs_from_ocsp(ocsps)))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jdoe/project-x86/lib/python3.11/site-packages/pyhanko/sign/validation/dss.py", line 159, in _embed_certs_from_ocsp
    return [self._embed_cert(cert_) for cert_ in extra_certs()]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jdoe/project-x86/lib/python3.11/site-packages/pyhanko/sign/validation/dss.py", line 159, in <listcomp>
    return [self._embed_cert(cert_) for cert_ in extra_certs()]
            ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jdoe/project-x86/lib/python3.11/site-packages/pyhanko/sign/validation/dss.py", line 171, in _embed_cert
    generic.StreamObject(stream_data=cert.dump())
                                     ^^^^^^^^^^^
  File "/Users/jdoe/project-x86/lib/python3.11/site-packages/asn1crypto/core.py", line 4121, in dump
    self._set_contents(force=force)
  File "/Users/jdoe/project-x86/lib/python3.11/site-packages/asn1crypto/core.py", line 3671, in _set_contents
    child_dump = child.dump(force=force)
                 ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jdoe/project-x86/lib/python3.11/site-packages/asn1crypto/core.py", line 4121, in dump
    self._set_contents(force=force)
  File "/Users/jdoe/project-x86/lib/python3.11/site-packages/asn1crypto/core.py", line 3671, in _set_contents
    child_dump = child.dump(force=force)
                 ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jdoe/project-x86/lib/python3.11/site-packages/asn1crypto/core.py", line 4589, in dump
    self._set_contents(force=force)
  File "/Users/jdoe/project-x86/lib/python3.11/site-packages/asn1crypto/core.py", line 4461, in _set_contents
    contents.write(child.dump(force=force))
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jdoe/project-x86/lib/python3.11/site-packages/asn1crypto/core.py", line 4121, in dump
    self._set_contents(force=force)
  File "/Users/jdoe/project-x86/lib/python3.11/site-packages/asn1crypto/core.py", line 3671, in _set_contents
    child_dump = child.dump(force=force)
                 ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jdoe/project-x86/lib/python3.11/site-packages/asn1crypto/core.py", line 2883, in dump
    native = self.parsed.dump(force=force)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jdoe/project-x86/lib/python3.11/site-packages/asn1crypto/core.py", line 4589, in dump
    self._set_contents(force=force)
  File "/Users/jdoe/project-x86/lib/python3.11/site-packages/asn1crypto/core.py", line 4461, in _set_contents
    contents.write(child.dump(force=force))
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jdoe/project-x86/lib/python3.11/site-packages/asn1crypto/core.py", line 4121, in dump
    self._set_contents(force=force)
  File "/Users/jdoe/project-x86/lib/python3.11/site-packages/asn1crypto/core.py", line 3671, in _set_contents
    child_dump = child.dump(force=force)
                 ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jdoe/project-x86/lib/python3.11/site-packages/asn1crypto/core.py", line 1721, in dump
    self.set(native)
  File "/Users/jdoe/project-x86/lib/python3.11/site-packages/asn1crypto/core.py", line 3100, in set
    part = int(part)
           ^^^^^^^^^
ValueError: invalid literal for int() with base 10: ''

**Environment (please complete the following information):**
 - OS: Os X Ventura

**Comment by MatthiasValvekens** [comment]
Hi @GaelC92,

That's an error bubbling up from `asn1crypto`, not pyHanko. More likely than not, it's a manifestation of this bug in FreeTSA's OCSP implementation: https://github.com/MatthiasValvekens/pyHanko/discussions/60.

TL;DR: FreeTSA, and especially its OCSP responder, is kinda crappy. Plus, it's not on any notable trust lists. See linked thread for an alternative TSA that is free to use _and_ actually trusted by PDF viewers in the wild :).

**Comment by GaelC92** [comment]
Thanks, I will switch :) 

In the mean time, I think I found the root cause of the error : the freetsa.org cacert has an empty line after X509v3 CRL Distribution Points, as shown by openssl : 
<img width="990" alt="image" src="https://user-images.githubusercontent.com/23708645/224813038-8dc0d643-d4fd-40e1-a91d-934d43016352.png">

**Comment by MatthiasValvekens** [comment]
Yeah, the empty CRL DistributionPoint name is wacky too, but the stack trace shows that it's coming from something OCSP related, so I'm like 85% sure it's the same bug as the one discussed in #60.

**Comment by GaelC92** [comment]
Yes, probably the same bug, I agree.... And apologize for the re-submit. 
Nonetheless, the empty value might well be involved in the OCSP wackyness...
